pandas/io/excel/_base.py

```diff
@@ -336,10 +336,10 @@ def read_excel(
 
 
 class _BaseExcelReader(metaclass=abc.ABCMeta):
-    def __init__(self, filepath_or_buffer):
+    def __init__(self, filepath_or_buffer, session=None):
```

Comment:
I'm a bit lost on how to pass the `session` object to this class, right now it `defaults` to `None` b/c it does not get the value from `read_excel`.